odoo/addons/base/models/res_partner.py

```diff
@@ -536,7 +536,7 @@ def write(self, vals):
         if vals.get('active') is False:
             for partner in self:
                 if partner.active and partner.user_ids:
-                    raise ValidationError(_('You cannot archive a contact linked to an internal user.'))
+                    raise ValidationError(_('You cannot archive a contact linked to a portal user.'))
```

Comment:
This message can be applied either for internal user or portal user, so I guess using portal user in message in not 100% correct.

Reply:
Should it say " a portal or internal user " then?

Reply:
Yes ```You cannot archive a contact linked to a portal or internal user```